osu.Game.Rulesets.Osu/Mods/OsuModSnapColour.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.ObjectExtensions;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Screens.Edit;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    /// <summary>
+    /// Mod that colours <see cref="HitObject"/>s based on the musical division they are on
+    /// </summary>
+    public class OsuModSnapColour : ModSnapColour, IApplicableToBeatmap, IApplicableToDrawableHitObject
+    {
+        private readonly OsuColour colours = new OsuColour();
+
+        [Resolved]
+        private IBeatmap? currentBeatmap { get; set; }
```

Comment:
This `[Resolved]` does nothing. As you [correctly pointed out](https://github.com/ppy/osu/pull/23520#discussion_r1192926111), this dependency injection mechanism only works for drawables. So the attribute should be removed as it's misleading at best.

Reply:
Thanks, I have removed the resolved tags